jupyter_ai_personas/pr_review_persona/ci_tools.py

```diff
@@ -0,0 +1,126 @@
+                        
+
+                        headers = {
+                            "Accept": "application/vnd.github+json",
+                            "Authorization": f"Bearer {self.github_token}",
+                            "X-GitHub-Api-Version": "2022-11-28"
+                        }
+                        log_url = f"https://api.github.com/repos/{repo_name}/actions/jobs/{job_id}/logs"
+                        log_response = requests.get(log_url, headers=headers)
+                        
+                        if log_response.status_code != 200:
+                            raise Exception(f"Failed to fetch logs: {log_response.status_code} {log_response.text} from {log_url}")
+                        log_content = log_response.text
+
+                        ##If seeing ThrottlingException in test aws account uncomment these lines [81-87] and line  92 AND comment line 93
+
+                        # # Extract key error lines from the log
+                        # log_lines = log_content.splitlines()
+                        # error_lines = []
+                        # for line in log_lines[-20:]:  
+                        #     if 'error:' in line.lower() or 'fail:' in line.lower():
+                        #         error_lines.append(line)
+                        #         if len(error_lines) >= 10: 
+                        #             break
+                        
+                        failure_data = {
+                            "name": job.name,
+                            "id": job_id,
+                            # "error_lines": error_lines if error_lines else [log_lines[-1]],  
+                            "log": log_content
+                        }
+                        failures.append(failure_data)
+
+                        if agent.session_state is None:
+                            agent.session_state = {}
+                        if "ci_logs" not in agent.session_state:
+                            agent.session_state["ci_logs"] = []
+                        
+                        agent.session_state["ci_logs"].append(failure_data)
+
+        print(f"Found {len(failures)} failed jobs")
+        return failures
+
+    def get_ci_logs(self, agent: Agent, job_name: str = None) -> list:
+        """
+        Retrieve CI failure logs from agent's session state.
+        
+        Args:
+            agent (Agent): The agent instance to access session state
+            job_name (str, optional): Filter logs by job name
+            
+        Returns:
+            list: List of failure logs matching the criteria
+        """
+        # Handle None session_state
+        if agent.session_state is None or "ci_logs" not in agent.session_state:
+            return []
+            
+        logs = agent.session_state["ci_logs"]
+        if job_name:
+            logs = [log for log in logs if log["name"] == job_name]
+            
+        return logs
```

Comment:
This seems redundant, as all the logs are already collected in `fetch_ci_failure_data` and this function is just returning the data from the session_state.